WeScan/Scan/RectangleFeaturesFunnel.swift

```diff
@@ -86,10 +97,16 @@ final class RectangleFeaturesFunnel {
             return
         }
         
-        if let previousRectangle = currentRectangle,
-            bestRectangle.rectangleFeature.isWithin(matchingThreshold, ofRectangleFeature: previousRectangle) {
+        if let previousRectangle = currentRectangle, bestRectangle.matchingScore >= minNumberOfMatches && bestRectangle.rectangleFeature.isWithin(10.0, ofRectangleFeature: previousRectangle) {
```

Comment:
Why use 10 instead of the previous value?

Reply:
I investigated this and made a few changes. 

1. I removed the condition as suggested, and found it made no difference to the scanning
2. I made a new variable, `autoScanMatchingThreshold`, to better document the "arbitrary" value I was using.
    2a. This variable (also documented in the commit) is what the RectangleFeaturesFunnel uses to determine if a rectangle is close enough to the previous one to be counted as an "auto scannable rectangle" - if it's not, it's shown but doesn't count for the Auto Scan count.
    2b. A higher value means the auto scan is quicker, but the rectangle will be less accurate. On the other hand, the lower the value, the longer it'll take for the auto scan, but it'll be way more accurate

I've additionally changed the threshold to 4.0 as I found it actually makes the auto scan *way* more accurate.